types.cc

```diff
@@ -694,6 +694,32 @@ void write_collection_value(bytes::iterator& out, cql_serialization_format sf, b
     out = std::copy_n(val_bytes.begin(), val_bytes.size(), out);
 }
 
+// Passing the wrong integer type to a generic serialization function is a particularly
+// easy mistake to do, so we want to disable template parameter deduction here.
+// Hence std::type_identity.
+template<typename T>
+void write_simple(bytes_ostream& out, std::type_identity_t<T> val) {
+    auto val_be = net::hton(val);
+    auto val_ptr = reinterpret_cast<const bytes::value_type*>(&val_be);
+    out.write(bytes_view(val_ptr, sizeof(T)));
+}
+
+void write_collection_value(bytes_ostream& out, cql_serialization_format sf, data::value_view val) {
```

Comment:
It is also possible to template `write_collection_value` on `OutputIterator` and support both `bytes::iterator` and `bytes_ostream::output_iterator`. Maybe not worth it is slower than this solution.

Reply:
I expect the other write_collection_value() will just die.